packages/react-dom/src/server/ReactPartialRendererContext.js

```diff
@@ -75,26 +75,50 @@ export function processContext(
   threadID: ThreadID,
 ) {
   const contextType = type.contextType;
-  if (typeof contextType === 'object' && contextType !== null) {
-    if (__DEV__) {
-      const isContextConsumer =
+  if (__DEV__) {
+    if ('contextType' in (type: any)) {
+      let isValid =
+        contextType !== null &&
+        contextType !== undefined &&
         contextType.$$typeof === REACT_CONTEXT_TYPE &&
-        contextType._context !== undefined;
-      if (contextType.$$typeof !== REACT_CONTEXT_TYPE || isContextConsumer) {
-        let name = getComponentName(type) || 'Component';
-        if (!didWarnAboutInvalidateContextType[name]) {
-          didWarnAboutInvalidateContextType[name] = true;
-          warningWithoutStack(
-            false,
-            '%s defines an invalid contextType. ' +
-              'contextType should point to the Context object returned by React.createContext(). ' +
-              'Did you accidentally pass the Context.%s instead?',
-            name,
-            isContextConsumer ? 'Consumer' : 'Provider',
-          );
+        contextType._context === undefined; // Not a <Context.Consumer>
+
+      if (!isValid && !didWarnAboutInvalidateContextType.has(type)) {
+        didWarnAboutInvalidateContextType.add(type);
+
+        let addendum = '';
+        if (contextType === null) {
```

Comment:
Sometimes it is nice to be able to conditionally set it:

```js
class Foo {
  static contextType = isServerEnvironment ? null : Context;
}
```

This provides no way to do that while also ensuring you stay declarative.

Is `null` really a common mistake? I'm guessing not since it's the only one you don't have a descriptive message for.

Reply:
Nope. Agree let’s omit it.